extensions/component/ui/use-component-query.ts

```diff
@@ -1,45 +1,88 @@
+import { useMemo, useEffect } from 'react';
 import { useDataQuery } from '@teambit/ui';
 import { gql } from 'apollo-boost';
-import { useMemo } from 'react';
 import { ComponentModel } from './component-model';
 import { ComponentError } from './component-error';
 
+const componentFields = gql`
+  fragment componentFields on Component {
+    id {
+      name
+      version
+      scope
+    }
+    displayName
+    server {
+      env
+      url
+    }
+    compositions {
+      identifier
+    }
+    tags {
+      version
+      snap {
+        hash
+        timestamp
+        message
+      }
+    }
+  }
+`;
+
 const GET_COMPONENT = gql`
   query Component($id: String!, $extensionId: String!) {
     getHost(id: $extensionId) {
       get(id: $id) {
-        id {
-          name
-          version
-          scope
-        }
-        displayName
-        server {
-          env
-          url
-        }
-        compositions {
-          identifier
-        }
-        tags {
-          version
-          snap {
-            hash
-            timestamp
-            message
-          }
-        }
+        ...componentFields
+      }
+    }
+  }
+  ${componentFields}
+`;
+
+const SUB_COMPONENT = gql`
+  subscription OnComponentChanged {
```

Comment:
adding subscription to component data. this will update the component model when compositions change